editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,261 @@
+ */
+public class AdvancedBeanPropertiesSection extends FusePropertySection {
+
+	/**
+	 * 
+	 * @param folder
+	 */
+	@Override
+	protected void createContentTabs(CTabFolder folder) {
+		List<Parameter> props = new ArrayList<>();
+		List<String> tabsToCreate = new ArrayList<>();
+		tabsToCreate.add(GROUP_COMMON);
+		
+		// define the properties we're handling here
+		Parameter idParam = BeanConfigUtil.createParameter(CamelBean.PROP_ID, String.class.getName());
+		idParam.setRequired("true");
+		props.add(idParam);
+		Parameter classParam = BeanConfigUtil.createParameter(CamelBean.PROP_CLASS, String.class.getName());
+		classParam.setRequired("true");
+		props.add(classParam);
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_SCOPE, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_DEPENDS_ON, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_INIT_METHOD, String.class.getName()));
+		props.add(BeanConfigUtil.createParameter(CamelBean.PROP_DESTROY_METHOD, String.class.getName()));
+
+		props.sort(new ParameterPriorityComparator());
+
+		for (String group : tabsToCreate) {
+			CTabItem contentTab = new CTabItem(this.tabFolder, SWT.NONE);
+			contentTab.setText(Strings.humanize(group));
+
+			Composite page = this.toolkit.createComposite(folder);
+			page.setLayout(new GridLayout(4, false));
+
+			generateTabContents(props, page);
+
+			contentTab.setControl(page);
+
+			this.tabs.add(contentTab);
+		}
+	}
+
+	private AbstractTextFieldParameterPropertyUICreator createTextFieldWithClassBrowse(final Parameter p, final Composite page) {
+		AbstractTextFieldParameterPropertyUICreator txtFieldCreator = 
+				new AttributeTextFieldPropertyUICreatorWithBrowse(dbc, modelMap, eip, selectedEP, p, page, getWidgetFactory());
+		txtFieldCreator.create();
+		createClassBrowseButton(page, txtFieldCreator.getControl());
+		return txtFieldCreator;
+	}
+
+	private AbstractTextFieldParameterPropertyUICreator createTextFieldWithMethodBrowse(final Parameter p, final Composite page) {
+		AbstractTextFieldParameterPropertyUICreator txtFieldCreator = 
+				new AttributeTextFieldPropertyUICreatorWithBrowse(dbc, modelMap, eip, selectedEP, p, page, getWidgetFactory());
+		txtFieldCreator.create();
+		createMethodBrowseButton(page, txtFieldCreator.getControl());
+		return txtFieldCreator;
+	}
+
+	/**
+	 * 
+	 * @param props
+	 * @param page
+	 * @param ignorePathProperties
+	 * @param group
+	 */
+	protected void generateTabContents(List<Parameter> props, final Composite page) {
+		props.sort(new ParameterPriorityComparator());
+		for (Parameter p : props) {
+			createPropertyLabel(toolkit, page, p);
+			createPropertyFieldEdition(page, p);
+		}
+	}
+
+	private void createPropertyFieldEdition(final Composite page, Parameter p) {
+		IValidator validator = null;
+		AbstractTextFieldParameterPropertyUICreator txtFieldCreator = null;
+
+		if (p.getName().contentEquals(CamelBean.PROP_CLASS)) {
```

Comment:
As p.getName() and CamelBean.PROP_CLASS are Strings, maybe better to use equals(), it will avoid the String.contentEquals to check for AbstractStringBuilder

(based on the implentation of String:

  ```
 public boolean contentEquals(CharSequence cs) {
        // Argument is a StringBuffer, StringBuilder
        if (cs instanceof AbstractStringBuilder) {
            if (cs instanceof StringBuffer) {
                synchronized(cs) {
                   return nonSyncContentEquals((AbstractStringBuilder)cs);
                }
            } else {
                return nonSyncContentEquals((AbstractStringBuilder)cs);
            }
        }
        // Argument is a String
        if (cs instanceof String) {
            return equals(cs);
        }
```

or am i missing something?